pandas/io/feather_format.py

```diff
@@ -0,0 +1,108 @@
+from pandas.compat import range
+from pandas.lib import infer_dtype
+
+
+def _try_import():
+    # since pandas is a dependency of feather
+    # we need to import on first use
+
+    try:
+        import feather
+    except ImportError:
+
+        # give a nice error message
+        raise ImportError("the feather-format library is not installed\n"
+                          "you can install via conda\n"
+                          "conda install feather-format -c conda-forge")
+    return feather
+
+
+def to_feather(df, path):
+    """
+    Write a DataFrame to the feather-format
+
+    Parameters
+    ----------
+    df : DataFrame
+    path : string
+        File path
+    """
+    if not isinstance(df, DataFrame):
+        raise ValueError("feather only support IO with DataFrames")
+
+    feather = _try_import()
+    valid_types = {'string', 'unicode'}
+
+    # validate index
+    # --------------
+
+    # validate that we have only a default index
+    # raise on anything else as we don't serialize the index
+
+    if not isinstance(df.index, (RangeIndex, Int64Index)):
+        raise ValueError("feather does not serializing {} "
+                         "for the index; you can .reset_index()"
+                         "to make the index into column(s)".format(
+                             type(df.index)))
+
+    if not df.index.equals(RangeIndex.from_range(range(len(df)))):
+        raise ValueError("feather does not serializing a non-default index "
+                         "for the index; you can .reset_index()"
+                         "to make the index into column(s)")
+
+    # validate columns
+    # ----------------
+
+    # must have unique column names
+    if not df.columns.is_unique:
+        raise ValueError("feather does not support duplicate columns")
+
+    # must be a Index
+    if isinstance(df.columns, MultiIndex):
+        raise ValueError("feather does not serializing a MultiIndex "
+                         "for the columns")
+
+    # must have value column names (strings only)
+    if df.columns.inferred_type not in valid_types:
+        raise ValueError("feather must have string column names")
+
+    # validate dtypes
+    # ---------------
+
+    # validate that we do not have any non-string object dtypes
+    # as these 'work', but will not properly de-serialize
+    objects = df.select_dtypes(include=['object'])
+    dtypes = [infer_dtype(o) for c, o in objects.iteritems()]
+    if len(set(dtypes) - valid_types):
+        invalid = DataFrame([[i, c, dtype] for i, (c, dtype) in
+                             enumerate(zip(objects.columns, dtypes))])
+        invalid.columns = ['ncolumn', 'column', 'inferred_dtype']
+        invalid = invalid[~invalid.inferred_dtype.isin(list(valid_types))]
+
+        msg = ("The following columns are not supported to serialize "
+               "to the feather-format:\n\n"
+               "{}".format(invalid.to_string()))
+        raise ValueError(msg)
```

Comment:
I think this whole function should be pushed down into Feather itself, and the dtype inference can be handled there too